wrappers/mmapwrapper.h

```diff
@@ -73,6 +75,12 @@ namespace HL {
     enum { Size = 8 * 1024UL };
     enum { Alignment = 64 * 1024UL };
 
+#elif defined(HL_APPLE_SILICON)
+    // macOS on Apple Silicon aligns 16K pages to a 16K boundary.
+    // FIXME: is this the correct alignment?
+    enum { Size = 16 * 1024UL };
+    enum { Alignment = 16 * 1024UL };
```

Comment:
I wasn't able to find any documentation about the alignment for Apple Silicon pages. I'm assuming they're 16K aligned?

Reply:
I believe this is correct. I'm not sure if `vm_page_size` is `constexpr` but if so, we could use that instead of hardcoding the numbers. Anyway, probably should also put in an assertion (if one is not there already) in the call to `mmap` to verify alignment and size.